src/utils/request/parseIsomorphicRequest.ts

```diff
@@ -15,7 +16,8 @@ export function parseIsomorphicRequest(
     url: request.url,
     method: request.method,
     body: parseBody(request.body, request.headers),
-    headers: request.headers,
+    credentials: request.credentials,
+    headers: new Headers(Array.from(request.headers.entries())),
```

Comment:
I think `request.headers` is already an instance of the `Headers` polyfill. Could you share more info as to why we need to construct it again? Are some headers being omitted? If so, have you understand why?

Reply:
To be honest, I don't understand perfectly the reason for this but it's needed to keep the `headers` field. Even if we call parseIsomorphicRequest multiple times.

I'll check how to disappear the headers field when calling parseIsomorphicRequest multiple times.

Reply:
Oh. I understood the reason. That the reason why is the `setRequestCookies` will overwrite the `headers` field.

Reply:
No worries!
The `headers` must not disappear—that's for certain. I just think that we don't need to deconstruct an existing `Headers` instance to construct it again 🤔 It'd be great to catch the issue you're experiencing into a unit test.

Reply:
I've considered it.

### The `parseIsomorphicRequest` always attaches the all cookies

This fact is commented in https://github.com/mswjs/msw/pull/1155/files#diff-738c9fc8b1eee13315e077f1aed118cc90af2e9cfb3965436290ff67a6d64284R33 .

I think it's correct so I don't modify it.

### The `setRequestCookie` consumes the `cookie` headers

So we should re-create the new `headers` field. (because it's overwritten).


Finally, I think current PR is correct.